### ISSUE TITLE
Wrong commit message & multiple tags on each commit

### ISSUE THREAD

**Issue by yashpola** [nus-cs2103-AY2324S2/forum#27]
Hi, I have the following 2 queries: 

1. Is it an issue if a commit has the wrong **message**? I.e. "Level 5. Automated Text UI Testing" when this is not one of the levels.

2. After pushing tags I noticed newer tags are stacked onto previous commits for past increments. So now my Level-0 commit has tags Level-5 to 0. Is this a major issue for the grader and how do I rectify it?

Thank you!

**Comment by damithc** [comment]
@yashpola It is normal for student to make mistakes when learning new Git operations. We will not penalize such mistakes. Try to figure out what you did wrong and avoid the same mistake in future. You are welcome to try to rectify them as well. 
That said, modifying past commits change their timestamp, which can affect our scripts that monitor student progress based on commit timestamps. So, best not to modify commits that are older than a 1-2 days.

> * Is it an issue if a commit has the wrong **message**? I.e. "Level 5. Automated Text UI Testing" when this is not one of the levels.

It will not affect grading, but you are welcome to try to rectify it (it's good to know how to modify a commit message later)

> * After pushing tags I noticed newer tags are stacked onto previous commits for past increments. So now my Level-0 commit has tags Level-5 to 0. Is this a major issue for the grader and how do I rectify it?

Perhaps you can share a screenshot of how the commit graph looks like? That can help others get a better understanding of the current situation.
In any case, this too will not affect grading.

**Comment by yashpola** [comment]
Thanks prof, is this the commit graph you mean?

<img width="1006" alt="image" src="https://github.com/nus-cs2103-AY2324S2/forum/assets/105275750/b8819da6-307b-4cf9-a8ea-e67005bbd11d">

**Comment by damithc** [comment]
@yashpola Let's see if any classmate can offer an explanation or some pointers. In the meantime, let it be (i.e., there is no need for any urgent action).

**Comment by tim-pipi** [comment]
@yashpola Hey there, if you are still interested in the methods to change your commit messages, do check out this helpful post on StackOverflow on [Changing Git Commit Message](https://stackoverflow.com/questions/8981194/changing-git-commit-message-after-push-given-that-no-one-pulled-from-remote).

Here is a TLDR Version:
**TL;DR: Changing a Git Commit Message**

- **Most Recent Commit (Not Pushed):** 
  - Run `git commit --amend`.
  - Edit the message in the text editor and save it.

- **Most Recent Commit (Already Pushed):** 
  - Amend the message using `git commit --amend`.
  - Force push using `git push --force` or `git push --force-with-lease`.

- **Older Commit:**
  - Start interactive rebase: `git rebase -i HEAD~n` (replace `n` with number of commits back).
  - Change `pick` to `reword` for the target commit in the list.
  - Save, then edit the message when prompted, and save again.
  - Complete the rebase and force push as above.

**Note:** Force pushing can disrupt others' work if they have based their work on the changed commits. Always coordinate with your team before rewriting pushed commit history. (Although this may not apply to you right now since you are still working on your Individual Project).

On a separate note, do try to follow the [Git Conventions](https://se-education.org/guides/conventions/git.html) as well (with regards to Commit Messages) to ensure clarity and consistency!

**Comment by yashpola** [comment]
@tim-pipi thanks for the help!

**Comment by SPWwj** [comment]
Hi @yashpola, I'm from your senior batch and wanted to drop in with a hint that might help with your issue.

For your IP project, all the relevant tags are listed in this [link](https://github.com/yashpola/ip/tags). I recommend going through them to ensure your tasks are correctly aligned with the respective commits.

Commit tagging is implemented for a specific purpose. It's important to familiarize yourself with this process. Understanding the workflow and objectives will provide you with a clearer overview, aiding in verifying that you're on the right track. Also, feel free to delve into various Git operations; they can be very insightful.

I hope you find this helpful. Good luck!

**Comment by damithc** [comment]
> This is how the tags stack up on github for a commit:
> <img alt="image" width="461" src="https://private-user-images.githubusercontent.com/105275750/298381917-d6a25dd7-0f63-4114-8a9b-b217e2f9cb9d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYwNjMzNzAsIm5iZiI6MTcwNjA2MzA3MCwicGF0aCI6Ii8xMDUyNzU3NTAvMjk4MzgxOTE3LWQ2YTI1ZGQ3LTBmNjMtNDExNC04YTliLWIyMTdlMmY5Y2I5ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDEyNFQwMjI0MzBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hNGYyZjM3ZWJjYjAzMDUyZDQyZDhhNjA1NWE1ZDkxZWFiMjM4MWZjNmQ0ZDNlZDIyMzg1NjE0NWQ2YmQyMTlkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.nY9l0v543resLpTLKhHjUX1_bAKVm3-UQEkeuwEaTL0">

@yashpola This seems to be a GitHub feature that shows all tagged commits that are descendants of the current commit (e.g., if one takes the code at commit tagged `Level-1`, that person would get the current commit as well. So, there is nothing wrong.

**Comment by yashpola** [comment]
@damithc oh understood prof, thanks! ill close this issue now
